spi/policy-monitor/policy-monitor-spi/src/main/java/org/eclipse/edc/connector/policy/monitor/spi/PolicyMonitorStore.java

```diff
@@ -0,0 +1,20 @@
+/*
+ *  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.connector.policy.monitor.spi;
+
+import org.eclipse.edc.spi.persistence.StateEntityStore;
+
+public interface PolicyMonitorStore extends StateEntityStore<PolicyMonitorEntry> {
```

Comment:
Do we need a dedicated store for that? 
Can't we just browse the active `ContractAgreement` (i.e. not expired nor canceled) stored in the contract negotiation store?

Reply:
we need the state machine for reliability, scalability and performances (the reason was detailed in the [related DR](https://github.com/eclipse-edc/Connector/tree/main/docs/developer/decision-records/2023-09-07-policy-monitor))

furthermore, `ContractAgreement`s don't have an "expired" or "canceled" state, is a condition that needs to be verified on the policy, so we'll need to query every `ContractAgreement` every time as there's no way to mark them "expired" nor "canceled".